src/main/java/uk/gov/hmcts/ccd/domain/service/getcasedocument/GetCaseDocumentOperation.java

```diff
@@ -0,0 +1,190 @@
+import uk.gov.hmcts.ccd.v2.external.domain.CaseDocument;
+import uk.gov.hmcts.ccd.v2.external.domain.CaseDocumentMetadata;
+import uk.gov.hmcts.ccd.v2.external.domain.Permission;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+@Service
+public class GetCaseDocumentOperation {
+
+    private final GetCaseOperation getCaseOperation;
+    private final CaseTypeService caseTypeService;
+    private final UserRepository userRepository;
+    private final CaseUserRepository caseUserRepository;
+    private final DocumentIdValidationService documentIdValidationService;
+
+    public static final String DOCUMENT_CASE_FIELD_CHILD_URL = "document_url";
+    public static final String DOCUMENT_CASE_FIELD_CHILD_NAME = "document_filename";
+    public static final String DOCUMENT_CASE_FIELD_CHILD_TYPE = "Document";
+    public static final String BAD_REQUEST_EXCEPTION_DOCUMENT_INVALID = "DocumentId is not valid";
+
+    @Autowired
+    public GetCaseDocumentOperation(
+        @Qualifier(CreatorGetCaseOperation.QUALIFIER) final GetCaseOperation getCaseOperation,
+        final CaseTypeService caseTypeService,
+        @Qualifier(CachedUserRepository.QUALIFIER) UserRepository userRepository,
+        @Qualifier(CachedCaseUserRepository.QUALIFIER) CaseUserRepository caseUserRepository,
+        DocumentIdValidationService documentIdValidationService) {
+
+        this.getCaseOperation = getCaseOperation;
+        this.caseTypeService = caseTypeService;
+        this.userRepository = userRepository;
+        this.caseUserRepository = caseUserRepository;
+        this.documentIdValidationService = documentIdValidationService;
+    }
+
+
+    public CaseDocumentMetadata getCaseDocumentMetadata(String caseId, String documentId) {
+
+        if (!documentIdValidationService.validateDocumentUUID(documentId)) {
+            throw new BadRequestException(BAD_REQUEST_EXCEPTION_DOCUMENT_INVALID);
+        }
+
+        final CaseDetails caseDetails = this.getCaseOperation.execute(caseId)
+            .orElseThrow(() -> new CaseNotFoundException(caseId));
+
+        if (!caseDetails.getReferenceAsString().isEmpty()) {
+            return CaseDocumentMetadata.builder()
+                .caseId(caseDetails.getReferenceAsString())
+                .caseTypeId(caseDetails.getCaseTypeId())
+                .jurisdictionId(caseDetails.getJurisdiction())
+                .document(getCaseDocument(caseDetails, documentId))
+                .build();
+        }
+        throw new CaseNotFoundException(caseId);
+    }
+
+    public CaseDocument getCaseDocument(CaseDetails caseDetails, String documentId) {
+
+        //retrieve the case data from details
+        Map<String, JsonNode> caseData = caseDetails.getData();
+
+        //get document field name
+        String documentField = getDocumentCaseField(caseData, documentId);
```

Comment:
if the case contains multiple fields for the same document id (technically possible and allowed in ccd backend), which field will be returned to here?